File: src/main/typescript/exercise1/MessageSigner.js
import { hash } from "./hash";

export class MessageSigner {
    constructor(secret, encodeFn = (secret, text) => hash(`!!!${secret}:::${text}!!!`)) {
        this.encodeFn = encodeFn;
        this.secret = secret;
    }

    sign(message) {
        return message.sign(this.secret, this.encodeFn);
    }
}

File: src/main/java/com/quatico/workshops/webcomponents/message/MessageSigner.java
package com.quatico.workshops.webcomponents.message;

public class MessageSigner {
    private final EncodeFn encodeFn;
    private final String secret;

    public MessageSigner(String secret) {
        this(secret, (String sec, String text) -> HashFn.apply(sec + text + sec));
    }

    public MessageSigner(String secret, EncodeFn encodeFn) {
        this.encodeFn = encodeFn;
        this.secret = secret;
    }

    public Message sign(Message message) {
        return message.sign(this.secret, this.encodeFn);
    }
}
